crates/client/db/tests/test_block.rs

```diff
@@ -0,0 +1,295 @@
+    assert_eq!(BLOCK_ID_0.resolve_db_block_id(backend).unwrap().unwrap(), BLOCK_ID_0);
+    assert_eq!(backend.get_block_info(&BLOCK_ID_0).unwrap().unwrap(), block.info);
+    assert_eq!(backend.get_block_inner(&BLOCK_ID_0).unwrap().unwrap(), block.inner);
+    assert_eq!(backend.get_block(&BLOCK_ID_0).unwrap().unwrap(), block);
+    assert_eq!(backend.get_block_n(&BLOCK_ID_0).unwrap().unwrap(), 0);
+    assert_eq!(backend.get_block_state_diff(&BLOCK_ID_0).unwrap().unwrap(), state_diff);
+}
+
+#[tokio::test]
+async fn test_store_pending_block() {
+    const BLOCK_ID_PENDING: DbBlockId = DbBlockId::Pending;
+
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    assert!(backend.get_block(&BLOCK_ID_PENDING).unwrap().is_none());
+
+    let block = block_pending_one();
+    let state_diff = state_diff_pending_one();
+
+    backend.store_block(block.clone(), state_diff.clone(), vec![]).unwrap();
+
+    assert!(backend.get_block_hash(&BLOCK_ID_PENDING).unwrap().is_none());
+    assert_eq!(backend.get_block_info(&BLOCK_ID_PENDING).unwrap().unwrap(), block.info);
+    assert_eq!(backend.get_block_inner(&BLOCK_ID_PENDING).unwrap().unwrap(), block.inner);
+    assert_eq!(backend.get_block(&BLOCK_ID_PENDING).unwrap().unwrap(), block);
+    assert_eq!(backend.get_block_state_diff(&BLOCK_ID_PENDING).unwrap().unwrap(), state_diff);
+}
+
+#[tokio::test]
+async fn test_erase_pending_block() {
+    const BLOCK_ID_PENDING: DbBlockId = DbBlockId::Pending;
+
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    backend.store_block(block_zero(), state_diff_zero(), vec![]).unwrap();
+    backend.store_block(block_pending_one(), state_diff_pending_one(), vec![]).unwrap();
+    backend.clear_pending_block().unwrap();
+
+    assert!(backend.get_block(&BLOCK_ID_PENDING).unwrap().is_none());
+
+    backend.store_block(block_one(), state_diff_one(), vec![]).unwrap();
+
+    let block = block_pending_two();
+    let state_diff = state_diff_pending_two();
+    backend.store_block(block.clone(), state_diff.clone(), vec![]).unwrap();
+
+    assert!(backend.get_block_hash(&BLOCK_ID_PENDING).unwrap().is_none());
+    assert_eq!(backend.get_block_info(&BLOCK_ID_PENDING).unwrap().unwrap(), block.info);
+    assert_eq!(backend.get_block_inner(&BLOCK_ID_PENDING).unwrap().unwrap(), block.inner);
+    assert_eq!(backend.get_block(&BLOCK_ID_PENDING).unwrap().unwrap(), block);
+    assert_eq!(backend.get_block_state_diff(&BLOCK_ID_PENDING).unwrap().unwrap(), state_diff);
+}
+
+#[tokio::test]
+async fn test_store_latest_block() {
+    let db = temp_db().await;
+    let backend = db.backend();
+
+    backend.store_block(block_zero(), state_diff_zero(), vec![]).unwrap();
+
+    let latest_block = block_one();
+    backend.store_block(latest_block.clone(), state_diff_one(), vec![]).unwrap();
+
+    assert_eq!(backend.get_latest_block_n().unwrap().unwrap(), 1);
+}
+
+#[tokio::test]
+async fn test_latest_confirmed_block() {
```

Comment:
the interesting thing to test here are the edge cases, we should probably fuzz-test these to make sure it works with "all" values, not for this PR but worth creating an issue if you agree

Reply:
I don’t know if this is very relevant for DB tests, but for other crates, absolutely